building documenation fails on develop branch

Building with  ``ENABLE_DOCS=ON`` at e7180035 fails with the following error

```
/scratch1/NCEPDEV/nems/Denise.Worthen/WORK/UFS_UTILS/sorc/global_cycle.fd/read_write_data.f90:1082: error: The following parameters of read_write_data::read_data(integer, intent(in) LSOIL, integer, intent(in) LENSFC, logical, intent(in) DO_NSST, logical, intent(out), optional IS_NOAHMP, character(len=50), intent(in), optional FNAME_INC, real, dimension(lensfc), intent(out), optional TSFFCS, real, dimension(lensfc, lsoil), intent(out), optional SMCFCS, real, dimension(lensfc), intent(out), optional SWEFCS, real, dimension(lensfc, lsoil), intent(out), optional STCFCS, real, dimension(lensfc), intent(out), optional TG3FCS, real, dimension(lensfc), intent(out), optional ZORFCS, real, dimension(lensfc), intent(out), optional CVFCS, real, dimension(lensfc), intent(out), optional CVBFCS, real, dimension(lensfc), intent(out), optional CVTFCS, real, dimension(lensfc, 4), intent(out), optional ALBFCS, real, dimension(lensfc), intent(out), optional VEGFCS, real, dimension(lensfc), intent(out), optional SLIFCS, real, dimension(lensfc), intent(out), optional CNPFCS, real, dimension(lensfc), intent(out), optional F10M, real, dimension(lensfc), intent(out), optional VETFCS, real, dimension(lensfc), intent(out), optional SOTFCS, real, dimension(lensfc, 2), intent(out), optional ALFFCS, real, dimension(lensfc), intent(out), optional USTAR, real, dimension(lensfc), intent(out), optional FMM, real, dimension(lensfc), intent(out), optional FHH, real, dimension(lensfc), intent(out), optional SIHFCS, real, dimension(lensfc), intent(out), optional SICFCS, real, dimension(lensfc), intent(out), optional SITFCS, real, dimension(lensfc), intent(out), optional TPRCP, real, dimension(lensfc), intent(out), optional SRFLAG, real, dimension(lensfc), intent(out), optional SNDFCS, real, dimension(lensfc), intent(out), optional VMNFCS, real, dimension(lensfc), intent(out), optional VMXFCS, real, dimension(lensfc, lsoil), intent(out), optional SLCFCS, real, dimension(lensfc, lsoil), intent(out), optional STCINC, real, dimension(lensfc, lsoil), intent(out), optional SLCINC, integer, intent(in), optional LSOIL_INCR, real, dimension(lensfc), intent(out), optional SLPFCS, real, dimension(lensfc), intent(out), optional ABSFCS, real, dimension(lensfc), intent(out), optional T2M, real, dimension(lensfc), intent(out), optional Q2M, real, dimension(lensfc), intent(out), optional SLMASK, real(kind=4), dimension(lsoil), intent(out), optional ZSOIL, type(nsst_data), optional NSST) are not documented:
  parameter 'fname_inc'
  parameter 'lsoil_incr' (warning treated as error, aborting now)
Exiting...
make[2]: *** [sorc/global_cycle.fd/docs/CMakeFiles/global_cycle_doc.dir/build.make:71: sorc/global_cycle.fd/docs/CMakeFiles/global_cycle_doc] Error 1
make[1]: *** [CMakeFiles/Makefile2:1213: sorc/global_cycle.fd/docs/CMakeFiles/global_cycle_doc.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

--- a/.github/PULL_REQUEST_TEMPLATE
+++ b/.github/PULL_REQUEST_TEMPLATE
@@ -19,6 +19,7 @@ If there are changes to the build or source code, the tests below must be conduc
 - [ ] Compile branch on all Tier 1 machines using Intel (Orion, Jet, Hera, Hercules and WCOSS2).
 - [ ] Compile branch on Hera using GNU.
 - [ ] Compile branch in 'Debug' mode on WCOSS2.
+- [ ] Compile with Doxygen on any machine with no errors.
 - [ ] Run unit tests locally on any Tier 1 machine.
 - [ ] Run relevant consistency tests locally on all Tier 1 machines.
 

--- a/.github/PULL_REQUEST_TEMPLATE
+++ b/.github/PULL_REQUEST_TEMPLATE
@@ -35,6 +35,10 @@ Add any links to pending PRs that are required prior to merging this PR. For exa
 ufs-community/UFS_UTILS/pull/<pr_number>
 
 ## DOCUMENTATION:
+All new and updated source code must be documented with Doxygen.
+
+- [ ] Doxygen is updated.
+
 If this PR is contributing new capabilities that need to be documented, please also include updates to the RST files in the docs/source directory as supporting material.
 
 ## ISSUE: 

--- a/sorc/global_cycle.fd/read_write_data.f90
+++ b/sorc/global_cycle.fd/read_write_data.f90
@@ -1036,6 +1036,7 @@ MODULE READ_WRITE_DATA
  !! @param[in] LENSFC Total number of points on a tile.
  !! @param[in] DO_NSST When true, nsst fields are read.
  !! @param[out] IS_NOAHMP When true, process for the Noah-MP LSM.
+ !! @param[in] FNAME_INC Name of the increment file.
  !! @param[out] TSFFCS Skin Temperature.
  !! @param[out] SMCFCS Total volumetric soil moisture.
  !! @param[out] SWEFCS Snow water equivalent.
@@ -1076,6 +1077,7 @@ MODULE READ_WRITE_DATA
  !! @param[out] NSST Data structure containing nsst fields.
  !! @param[in] SLCINC Liquid soil moisture increments on the cubed-sphere tiles
  !! @param[in] STCINC Soil temperature increments on the cubed-sphere tiles
+ !! @param[in] LSOIL_INCR -  Number of soil layers (from top) to apply soil increments to
  !! @author George Gayno NOAA/EMC
  !! @author Yuan Xue: add capability to read soil related increments on the
  !! cubed-sphere tiles directly

--- a/sorc/global_cycle.fd/read_write_data.f90
+++ b/sorc/global_cycle.fd/read_write_data.f90
@@ -1619,7 +1619,11 @@ MODULE READ_WRITE_DATA
  ELSE
      ! THIS IS A REGRIDDED GSI FILE
      IF (PRESENT(STCINC)) THEN
-     DO K = 1, LSOIL_INCR
+       IF (.NOT.PRESENT(LSOIL_INCR)) THEN
+         write(6,*)'FATAL ERROR variable lsoil_incr not declared.'
+         CALL MPI_ABORT(MPI_COMM_WORLD, 134, ERROR)
+       END IF
+       DO K = 1, LSOIL_INCR
          WRITE(K_CH, '(I1)') K
 
          INCVAR = "soilt"//K_CH//"_inc"
@@ -1630,10 +1634,14 @@ MODULE READ_WRITE_DATA
          CALL NETCDF_ERR(ERROR, 'READING soilt*_inc increments') 
 
          STCINC(:,K) = RESHAPE(dummy, (/LENSFC/))
-     ENDDO
+       ENDDO
      ENDIF
      IF (PRESENT(SLCINC)) THEN
-     DO K = 1, LSOIL_INCR
+       IF (.NOT.PRESENT(LSOIL_INCR)) THEN
+         write(6,*)'FATAL ERROR variable lsoil_incr not declared.'
+         CALL MPI_ABORT(MPI_COMM_WORLD, 136, ERROR)
+       END IF
+       DO K = 1, LSOIL_INCR
          WRITE(K_CH, '(I1)') K
 
          INCVAR = "slc"//K_CH//"_inc"
@@ -1644,7 +1652,7 @@ MODULE READ_WRITE_DATA
          CALL NETCDF_ERR(ERROR, 'READING slc*_inc increments') 
 
          SLCINC(:,K) = RESHAPE(dummy, (/LENSFC/))
-     ENDDO
+       ENDDO
      ENDIF
  ENDIF 
 

--- a/sorc/global_cycle.fd/read_write_data.f90
+++ b/sorc/global_cycle.fd/read_write_data.f90
@@ -1077,7 +1077,7 @@ MODULE READ_WRITE_DATA
  !! @param[out] NSST Data structure containing nsst fields.
  !! @param[in] SLCINC Liquid soil moisture increments on the cubed-sphere tiles
  !! @param[in] STCINC Soil temperature increments on the cubed-sphere tiles
- !! @param[in] LSOIL_INCR -  Number of soil layers (from top) to apply soil increments to
+ !! @param[in] LSOIL_INCR Number of soil layers (from top) to apply soil increments to
  !! @author George Gayno NOAA/EMC
  !! @author Yuan Xue: add capability to read soil related increments on the
  !! cubed-sphere tiles directly

